api_key_stamper/src/lib.rs

```diff
@@ -29,10 +38,22 @@ pub enum StamperError {
     HexDecode(String),
 }
 
+/// A stamped header to attach to an HTTP request.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct StampHeader {
+    pub name: &'static str,
+    pub value: String,
+}
+
+/// Generic stamper that operates over raw bytes and returns an HTTP header.
+pub trait Stamp: Send + Sync {
```

Comment:
Why `Send + Sync`? Feels a bit unnecessary (see these docs: https://doc.rust-lang.org/nomicon/send-and-sync.html)